newIDE/app/src/MainFrame/Preferences/PreferencesProvider.js

```diff
@@ -73,7 +74,8 @@ export default class PreferencesProvider extends React.Component<Props, State> {
     // Checking for updates is only done on Electron.
     // Note: This could be abstracted away later if other updates mechanisms
     // should be supported.
-    if (!ipcRenderer) return;
+    const { cmdArguments } = this.props;
+    if (!ipcRenderer || cmdArguments['disable-update-check']) return;
```

Comment:
As explained before, better pass everywhere a single boolean `disableCheckForUpdates `, will make dependency on this easier to see and easier to search in the codebase.
("where is the `disableCheckForUpdates` changing something?" "easy, just search `disableCheckForUpdates ` in the whole codebase VERSUS "oops there are tons of cmdArguments everywhere in the codebase, I don't understand what is being used - let's not touch it because I'm now too scared of breaking something" 😄)